### PR TITLE
Function expressions

### PR DESCRIPTION
**Pull request by otmon76**

### CODE REVIEW

File: 1-js/02-first-steps/16-function-expressions/article.md [hunk 1]
```diff
@@ -1,380 +1,381 @@
...
 ```js
-let sayHi = function() {
-  alert( "Hello" );
+let řekniAhoj = function() {
+  alert( "Ahoj" );
 };
 ```
 
-Here we can see a variable `sayHi` getting a value, the new function, created as `function() { alert("Hello"); }`.
+Zde je funkce vytvořena a výslovně přiřazena do proměnné jako kterákoli jiná hodnota. Nezáleží na tom, jak je funkce definována. Je to jen hodnota uložená do proměnné `řekniAhoj`.
```
Comment (danipoma) [inline 1-js/02-first-steps/16-function-expressions/article.md:25]:
```suggestion
Zde vidíme proměnnou `řekniAhoj`, do níž je uložena hodnota, kterou je nová funkce, vytvořená jako `function() { alert("Ahoj"); }`.
```

File: 1-js/02-first-steps/16-function-expressions/article.md [hunk 2]
```diff
@@ -1,380 +1,381 @@
...
 
-As the function creation happens in the context of the assignment expression (to the right side of `=`), this is a *Function Expression*.
+Význam obou částí kódu je stejný: „vytvořit funkci a uložit ji do proměnné `řekniAhoj`“.
```
Comment (danipoma) [inline 1-js/02-first-steps/16-function-expressions/article.md:27]:
```suggestion
Protože k vytvoření funkce dochází v kontextu přiřazovacího výrazu (na pravé straně `=`), je to *funkční výraz*.
```

File: 1-js/02-first-steps/16-function-expressions/article.md [hunk 3]
```diff
@@ -1,380 +1,381 @@
...
+  alert( "Ahoj" );
 }
 
-let func = sayHi;    // (2) copy
+let funkce = řekniAhoj;  // (2) zkopírování
 
-func(); // Hello     // (3) run the copy (it works)!
-sayHi(); // Hello    //     this still works too (why wouldn't it)
+funkce(); // Ahoj        // (3) spustíme kopii (funguje to!)
+řekniAhoj(); // Ahoj     //     tohle pořád funguje taky (proč by nemělo)
```
Comment (danipoma) [inline 1-js/02-first-steps/16-function-expressions/article.md:71]:
```suggestion
řekniAhoj(); // Ahoj     //     i tohle stále funguje (proč by nemělo)
```

File: 1-js/02-first-steps/16-function-expressions/article.md [hunk 4]
```diff
@@ -1,380 +1,381 @@
...
 
-The answer is simple: a Function Expression is created here as `function(…) {…}` inside the assignment statement: `let sayHi = …;`. The semicolon `;` is recommended at the end of the statement, it's not a part of the function syntax.
+Odpověď je jednoduchá: funkční výraz je zde vytvořen jako `function(…) {…}` uvnitř příkazu přiřazení: `let řekniAhoj = …;`. Středník `;` se doporučuje uvádět na konci příkazu, není to součást syntaxe funkce.
 
-The semicolon would be there for a simpler assignment, such as `let sayHi = 5;`, and it's also there for a function assignment.
+Středník by zde byl i u jednoduššího přiřazení, například `let řekniAhoj = 5;`, a je zde také u přiřazení funkce.
 ````
 
-## Callback functions
+## Callbackové funkce
```
Comment (danipoma) [inline 1-js/02-first-steps/16-function-expressions/article.md:111]:
```suggestion
## Funkce zpětného volaní (callback funkce)
```

File: 1-js/02-first-steps/16-function-expressions/article.md [hunk 5]
```diff
@@ -1,380 +1,381 @@
...
+Deklarace funkcí jsou odlišné.
 
-**A Function Declaration can be called earlier than it is defined.**
+**Deklarovaná funkce může být volána ještě dříve, než je definována.**
 
-For example, a global Function Declaration is visible in the whole script, no matter where it is.
+Například globální deklarace funkce je viditelná v celém skriptu, ať se nachází kdekoli.
 
-That's due to internal algorithms. When JavaScript prepares to run the script, it first looks for global Function Declarations in it and creates the functions. We can think of it as an "initialization stage".
+Je to způsobeno vnitřními algoritmy. Když JavaScript připravuje spuštění skriptu, nejprve v něm najde globální deklarace funkcí a tyto funkce vytvoří. Můžeme to považovat za „inicializační krok“.
```
Comment (danipoma) [inline 1-js/02-first-steps/16-function-expressions/article.md:219]:
```suggestion
Je to způsobeno vnitřními algoritmy. Když JavaScript připravuje spuštění skriptu, nejprve v něm najde globální deklarace funkcí a tyto funkce vytvoří. Můžeme to považovat za „přípravnou fázi“.
```

File: 1-js/02-first-steps/16-function-expressions/article.md [hunk 6]
```diff
@@ -1,380 +1,381 @@
...
-  welcome();               // /   (runs)
+  uvítání();               // /   (spustí se)
 */!*
 
 } else {
 
-  function welcome() {
-    alert("Greetings!");
+  function uvítání() {    
+    alert("Zdravíme vás!");
```
Comment (danipoma) [inline 1-js/02-first-steps/16-function-expressions/article.md:306]:
```suggestion
    alert("Dobrý den!");
```

**Review by danipoma — CHANGES_REQUESTED** [review]
Removed two paragraphs since they were not in source files anymore and also changed some text without suggestion since it was clear translation inconsistency. 

Please take a look on my suggestions.

**Comment by otmon76** [comment]
"Removed two paragraphs since they were not in source files anymore and also changed some text without suggestion since it was clear translation inconsistency."
this is true, I didn't notice that the two paragraphs were removed

**Comment by otmon76** [comment]
I am sorry I mistakenly merged it but the suggestions were applied.

**Pull request merged**